dotcom-rendering/src/lib/discussionApi.tsx

```diff
@@ -85,24 +89,40 @@ export const getDiscussion = async (
 
 	const url = joinUrl(options.baseUrl, 'discussion', shortUrl) + params;
 
-	const resp = await fetch(url, {
+	const json = (await fetch(url, {
 		headers: {
 			...options.headers,
 		},
-	});
-
-	const discussionResponse = parseDiscussionResponse(await resp.json());
-
-	return discussionResponse?.errorCode ===
-		'DISCUSSION_ONLY_AVAILABLE_IN_LINEAR_FORMAT'
-		? // We need force a refetch with unthreaded set, as we don't know
-		  // that this discussion is only available in linear format until
-		  // we get the response to tell us
-		  getDiscussion(shortUrl, {
-				...opts,
-				...{ threads: 'unthreaded' },
-		  })
-		: discussionResponse;
+	})
+		.then((r) => r.json())
+		.catch(() => undefined)) as unknown;
+
+	if (!json) return { kind: 'error', error: 'NetworkError' };
+
+	const result = safeParse(discussionApiResponseSchema, json);
```

Comment:
Perhaps we should do something like:

```ts
const getDiscussionJson = async (
	url: string,
	headers: typeof options['headers'],
): Promise<Result<'ApiError' | 'NetworkError', unknown>> => {
	try {
		const response = await fetch(url, { ...headers });

		return { kind: 'ok', value: await response.json() };
	} catch (error) {
		if (error instanceof SyntaxError) {
			return { kind: 'error', error: 'ApiError' };
		}

		return { kind: 'error', error: 'NetworkError' };
	}
}

const jsonResult = await getDiscussionJson(url, options.headers);

if (jsonResult.kind === 'error') return jsonResult;

const result = safeParse(discussionApiResponseSchema, jsonResult.value);
```

to distinguish JSON parsing errors from network errors?

Reply:
created a `fetchJSON` helper in 52e5c6ee2fb2f898d391743de09b0b1d4e7ed125